### PR TITLE
PLAT-782: functests refactoring

### PR DESCRIPTION
**Pull request by sexyhamster**
- functests cleanup
- timeouts for tests
- moved from syscall.SIGINT to os.Interrupt for windows usage
- little launchnet cleanup
- refactored benchmark runs

### CODE REVIEW

File: ledger-core/Makefile.testing [hunk 1]
```diff
@@ -25,7 +25,6 @@ test_with_coverage: ## run unit and slow tests with generation of coverage file
 
 .PHONY: test_func
 test_func: ## run functest FUNCTEST_COUNT times
-	pkill -9 insolard || echo "No insolard running" # dirty hack to terminate all insolard processes
```
Comment (ivanshibitov) [inline ledger-core/Makefile.testing:28]: it helped a lot =(

Comment (sexyhamster) [reply ledger-core/Makefile.testing:28]: I moved it to launchnet.sh where it should be

**Comment by ivanshibitov** [comment]
I think benchmarks on cloud will not work correctly, because nodes will not stop after every run

**Comment by sexyhamster** [comment]
Fixed benchmark runners

File: ledger-core/application/functest/main_func_cloud_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	launchnet.SetCloudFileLogging(true)
+
+	var (
+		numVirtual = 5
+		numLight   = 0
+		numHeavy   = 0
+		err        error
+	)
+	if numVirtualStr := os.Getenv("NUM_DISCOVERY_VIRTUAL_NODES"); numVirtualStr != "" {
```
Comment (ivanshibitov) [inline ledger-core/application/functest/main_func_cloud_test.go:39]: May be move it to function. It used twice here and in main_func_cloud_with_consensus_test.go

File: ledger-core/cmd/insolard/main.go [hunk 3]
```diff
@@ -81,7 +85,15 @@ func runInsolardServer(configPath, genesisConfigPath, roleString string) {
...
 	cfg := readConfig(configPath)
 	fmt.Printf("Starts with configuration:\n%s\n", configuration.ToString(&cfg))
-	s = server.NewNode(cfg)
+
+	signChan := make(chan os.Signal, 1)
+	signal.Notify(signChan, os.Interrupt, syscall.SIGTERM)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go stopper(cancel, signChan)
```
Comment (ivanshibitov) [inline ledger-core/cmd/insolard/main.go:94]: may be
```suggestion
        fun  registerStopper() context.Context {
	    signChan := make(chan os.Signal, 1)
 	    signal.Notify(signChan, os.Interrupt, syscall.SIGTERM)

            ctx, cancel := context.WithCancel(context.Background())

	    go stopper(cancel, signChan)
	    
	    return ctx
	}
```

File: ledger-core/application/testutils/launchnet/launchnet.go [hunk 4]
```diff
@@ -187,10 +192,6 @@ func GetDiscoveryNodesCount() (int, error) {
 }
 
 func GetNodesCount() (int, error) {
-	if isCloudMode() {
```
Comment (ivanshibitov) [inline ledger-core/application/testutils/launchnet/launchnet.go:190]: I think TestGetStatus will fail due to that

File: ledger-core/scripts/insolard/launchnet.sh [hunk 5]
```diff
@@ -252,7 +252,7 @@ headless=false
 
 process_input_params()
 {
-    echo "run arguments: $@"
+    echo "run arguments: $*"
```
Comment (ivanshibitov) [inline ledger-core/scripts/insolard/launchnet.sh:255]: hmm...
I think there is no different here =)

File: ledger-core/application/testutils/launchnet/launchnet.go (hunk 4 above)
Comment (sexyhamster) [reply ledger-core/application/testutils/launchnet/launchnet.go:190]: fixed

File: ledger-core/cmd/insolard/main.go (hunk 3 above)
Comment (sexyhamster) [reply ledger-core/cmd/insolard/main.go]: done

File: ledger-core/application/functest/main_func_cloud_test.go (hunk 2 above)
Comment (sexyhamster) [reply ledger-core/application/functest/main_func_cloud_test.go]: done

File: ledger-core/Makefile.testing [hunk 6]
```diff
@@ -25,26 +25,25 @@ test_with_coverage: ## run unit and slow tests with generation of coverage file
...
 	PULSARD_ONESHOT=$(PULSARD_ONESHOT) \
 	PULSARD_PULSAR_PULSETIME=$(PULSARD_PULSAR_PULSETIME) \
 	CGO_ENABLED=1 $(GOTEST) -test.v $(TEST_ARGS) -tags "functest copylogtxt $(FUNCTEST_BUILD_TAGS)" ./application/functest -count=$(FUNCTEST_COUNT) -args TESTLOG_OUT=$(EXTENDED_FUNCTIONAL_LOG) TESTLOG_MARKS=1 TESTLOG_TXT=1
 
 .PHONY: test_func_cloud
 test_func_cloud: ## run functest on insolard in cloud mode
-	PULSARD_ONESHOT=FALSE CLOUD_MODE=TRUE PULSARD_PULSAR_PULSETIME=2000 FUNCTEST_COUNT=100 $(MAKE) test_func
```
Comment (bigbes) [inline ledger-core/Makefile.testing:35]: please, don't remove PULSARD_ONESHOT

Comment (bigbes) [reply ledger-core/Makefile.testing:35]: Or it's controlled globally?

Comment (sexyhamster) [reply ledger-core/Makefile.testing:35]: it doesnot affect func cloud run

Comment (sexyhamster) [reply ledger-core/Makefile.testing:35]: why do we need PULSARD_ONESHOT=FALSE?

**Review by ita-sammann — APPROVED** [review]

**Review by ivanshibitov — APPROVED** [review]

**Pull request merged**